geospaas/base_viewer/tests.py

```diff
@@ -102,44 +103,25 @@ def test_the_get_verb_of_GUI(self):
-
-    def setUp(self):
-        self.v1 = IndexView()
-        self.ds = self.v1.get_all_datasets()
-        self.forms = self.v1.create_forms(self)
-
-    def test_creating_the_forms(self):
-        '''shall create the specified fields of forms'''
-        self.assertTrue('polygon' in self.forms.__str__())
-        self.assertTrue('time_coverage_start' in self.forms.__str__())
-        self.assertTrue('time_coverage_end' in self.forms.__str__())
-        self.assertTrue('source' in self.forms.__str__())
-
-    def test_setting_the_context_correctly(self):
-        '''shall return the correct key of context dictionary for correct viewing by templates'''
-        context = self.v1.set_context(self.forms, self.ds)
-        self.assertIn('TimeAndSourceForm', context.keys())
-        self.assertIn('SpatialSearchForm', context.keys())
-        self.assertIn('datasets', context.keys())
+class IndexViewTests(TestCase):
+    @patch('geospaas.base_viewer.views.CatalogDataset')
+    def test_get_all_datasets(self, mock_dataset):
+        """ Shall call CatalogDataset.objects.all() inside get_all_datasets """
+        IndexView.get_all_datasets()
+        mock_dataset.objects.all.assert_called_once()
 
     def test_get_filtered_datasets(self):
-        '''shall return the fixture datasets when large time-interval is set and none of them when a nonrelevant one is set'''
-        forms = self.v1.create_forms({'polygon': '',
-                                      'time_coverage_start': timezone.datetime(2000, 12, 29),
-                                      'time_coverage_end': timezone.datetime(2020, 1, 1),
-                                      'source': 1})
-        forms = self.v1.validate_forms(forms)
-        ds = self.v1.get_filtered_datasets(forms)
-        self.assertEqual(len(ds), 2)  # shall return the fixture datasets
-        self.assertEqual(
-            '<QuerySet [<Dataset: AQUA/MODIS/2010-01-01T00:00:00+00:00>, <Dataset: AQUA/MODIS/2010-01-02T00:00:00+00:00>]>', str(ds))
-        forms = self.v1.create_forms({'polygon': '',
-                                      'time_coverage_start': timezone.datetime(2019, 12, 29),
-                                      'time_coverage_end': timezone.datetime(2020, 1, 1),
-                                      'source': 1})
-        forms = self.v1.validate_forms(forms)
-        ds = self.v1.get_filtered_datasets(forms)
-        self.assertEqual(len(ds), 0)  # shall return no datasets
+        """ Shall  call filter function from form class once """
+        form = MagicMock()
+        ds = IndexView.get_filtered_datasets(form)
+        form.filter.assert_called_once()#form.filter.
+
+    def test_set_context(self):
+        """ Shall  contain 'OverallForm' and 'datasets' in the context.
+        Results should not be filtered by this function """
+        form = MagicMock()
+        ds = MagicMock()
+        context = IndexView.set_context(form,ds)
+        form.filter.assert_not_called()
```

Comment:
Not really needed here.
```suggestion
```

Reply:
I think it is better to have it here because this line makes sure that all datasets are shown and no one is filtered. If in the future development the filter function is called incorrectly, this line prevents it. It is needed to show a **_comprehensive_** result. Kindly, how do you think? Is it still not needed?

Reply:
OK, it is a minor thing anyway